sdk/storage/azure_storage_blob/tests/blob_client.rs

```diff
@@ -423,3 +425,91 @@ async fn test_leased_blob_operations(ctx: TestContext) -> Result<(), Box<dyn Err
     container_client.delete_container(None).await?;
     Ok(())
 }
+
+#[recorded::test]
+async fn test_set_legal_hold(ctx: TestContext) -> Result<(), Box<dyn Error>> {
+    //TODO: Need separate accounts, need a test account with versioning enabled
+    // Recording Setup
+    let recording = ctx.recording();
+    let container_client = get_container_client(recording, false).await?;
+    let blob_client = container_client.blob_client(get_blob_name(recording));
+    container_client.create_container(None).await?;
+    create_test_blob(&blob_client).await?;
+
+    // Set Legal Hold
+    blob_client.set_legal_hold(true, None).await?;
+    let response = blob_client.get_properties(None).await?;
+    // Assert
+    let legal_hold = response.legal_hold()?;
+    assert!(legal_hold.unwrap());
+
+    // Attempt Operation While Legal Hold Active
+    let response = blob_client.delete(None).await;
+    // Assert
+    let error = response.unwrap_err().http_status();
+    assert_eq!(StatusCode::Conflict, error.unwrap());
+
+    // Remove Legal Hold
+    blob_client.set_legal_hold(false, None).await?;
+    let response = blob_client.get_properties(None).await?;
+    // Assert
+    let legal_hold = response.legal_hold()?;
+    assert!(!legal_hold.unwrap());
+
+    // Even after removing the legal hold:
+    // Error: Error { context: Full(Custom { kind: HttpResponse { status: Conflict, error_code: Some("ContainerImmutableStorageWithVersioningEnabled") },
+    // error: HttpError { status: Conflict, details: ErrorDetails { code: Some("ContainerImmutableStorageWithVersioningEnabled"), message: None },
+    // body: <?xml version="1.0" encoding="utf-8"?><Error><Code>ContainerImmutableStorageWithVersioningEnabled</Code><Message>
+    // The requested operation is not allowed as the container has a immutable storage with versioning enabled.
+    // Seems like in Python testing, we need the management client in order to delete the container
```

Comment:
As called out here -- seems like we can't do a normal delete and need the management client as we did in Python.